### PR TITLE
Sign up/Login Feature

### PR DESCRIPTION
**Pull request by coo990**
Completed signup/login feature ticket with passing and refactored tests.

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class SessionsController < ApplicationController
+  skip_before_action :authorized, only: [:new, :create, :welcome]
+
+  def new
+  end
+
+  def create
```
Comment (Ashley-Slaney) [inline app/controllers/sessions_controller.rb:7]: Is this method for logging in or creating an account? Could use either comments or a more descriptive method name to make it clear

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1,7 +1,11 @@
+<h1>Welcome You are Logged In, <%= current_user.username %></h1>
```
Comment (Ashley-Slaney) [inline app/views/posts/index.html.erb:1]: It might read better as "Welcome <username>" rather than "Welcome You are Logged In, <username>"

File: app/views/sessions/create.html.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<h1>Sessions#create</h1>
```
Comment (Ashley-Slaney) [inline app/views/sessions/create.html.erb]: Is this file necessary?

File: app/views/sessions/login.html.erb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<h1>Sessions#login</h1>
```
Comment (Ashley-Slaney) [inline app/views/sessions/login.html.erb]: RE: sessions/create

File: app/views/sessions/page_requires_login.html.erb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<h1>You have access to this page, <%= current_user.username %></</h1>
```
Comment (Ashley-Slaney) [inline app/views/sessions/page_requires_login.html.erb]: What is happening on this page?

File: app/views/sessions/welcome.html.erb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<h1>Welcome</h1>
+<% if logged_in? %>
```
Comment (Ashley-Slaney) [inline app/views/sessions/welcome.html.erb:2]: If a user is logged in then maybe they could go straight into the posts page rather than having to click to view the posts

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (coo990) [reply app/controllers/sessions_controller.rb:7]: It was created from rails when we created the sessions controller, but we will clear them as they currently have no use.

File: app/views/sessions/create.html.erb (hunk 3 above)
Comment (coo990) [reply app/views/sessions/create.html.erb]: Again was created by rails when we ran the generate method for the sessions controller, it could be useful when we add CSS styles

File: app/views/sessions/page_requires_login.html.erb (hunk 5 above)
Comment (coo990) [reply app/views/sessions/page_requires_login.html.erb]: It's another page to confirm whether the user is logged in or not, probably not necessary moving forward.

File: app/views/sessions/welcome.html.erb (hunk 6 above)
Comment (coo990) [reply app/views/sessions/welcome.html.erb:2]: We actually did modify for it to do that I think from our latest commits after this.

File: config/routes.rb [hunk 7]
```diff
@@ -1,5 +1,12 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  root 'sessions#welcome'
```
Comment (Ashley-Slaney) [inline config/routes.rb:2]: Is there a way to redirect to a certain page based on whether the user is logged in/out? 
Example:
A user loads up the page and you check that they are logged in and automatically send them to the posts page rather than having them click the view posts button and something similar for when they are logged out and send them to the log in or sign up page

File: spec/features/user_signup_spec.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.feature "Sign up", type: :feature do
+  scenario "User can sign up" do
+    visit "/welcome"
+    click_button "Sign Up"
+    fill_in "Username", with: "JooMin"
```
Comment (Ashley-Slaney) [inline spec/features/user_signup_spec.rb]: JooMin spelt differently in this file compared to the others so could possibly cause confusion or issues later down the line when creating more tests?

**Review by Ashley-Slaney — COMMENTED** [review]
- Good commit process - clear commit messages and regular commits
- Most methods are easy to understand by the name
- Good feature test coverage
- Noticed you were using 'require 'rails_helper'' in a lot of spec files and wasn't sure where that is referring to - Let's check this later please

File: config/routes.rb (hunk 7 above)
Comment (joominchoi) [reply config/routes.rb:2]: It should be doing that now! But if not we can hop on a call later to quickly change where the page is redirected to depending on the login status

Comment (coo990) [reply config/routes.rb:2]: Yes, once the user logs in, it redirects straight to the posts page as it only has access if the user is logged in.

File: spec/features/user_signup_spec.rb (hunk 8 above)
Comment (coo990) [reply spec/features/user_signup_spec.rb]: Changed username field to keep it consistent
